app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java

```diff
@@ -0,0 +1,134 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import static android.content.ContentValues.TAG;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.HabitList;
+import com.example.habitsmasher.ItemAdapter;
+import com.example.habitsmasher.R;
+import com.example.habitsmasher.ui.home.HomeFragment;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.firebase.firestore.WriteBatch;
+
+import java.util.List;
+
+
+/**
+ * Custom adapter class that is used to connect the Firestore database and the
+ * RecyclerView displaying the habits
+ */
+public class HomeHabitItemAdapter extends ItemAdapter<Habit, HomeHabitItemAdapter.HabitViewHolder> {
+    private static HabitList _habits;
+    private final String _userId;
+    private Context _context;
+
+    /**
+     * Create a new RecyclerView adapter that listens to a Firestore Query.  See {@link
+     * FirestoreRecyclerOptions} for configuration options.
+     * @param options the firestore entities
+     * @param habits the habit list
+     * @param userId the username of the current user
+     */
+    public HomeHabitItemAdapter(@NonNull FirestoreRecyclerOptions<Habit> options,
+                            HabitList habits,
+                            String userId) {
+        super(options);
+        _habits = habits;
+        _userId = userId;
+    }
```

Comment:
This new adapter is needed to use the new home_habit_row I created. I found that you can't edit the row without making a new adapter, so that's what I did.